doc/reference/testing.rst

```diff
@@ -111,6 +111,90 @@ supported.
 
 .. _unittest documentation: https://docs.python.org/2/library/unittest.html
 
+Test tags
```

Comment:
I'd rename the section to "test selection", tags are just the mechanism underlying it.

Also maybe split it into 3 subsections e.g. "invocation" (command line bits when trying to run tests), "usage" (API to tag one's tests) and "built-in tags" or somesuch.